src/util/http_utils.js

```diff
@@ -248,7 +247,9 @@ function read_request_body(req, options) {
     });
 }
 
-const parse_xml_to_js = util.promisify(xml2js.parseString);
+function parse_xml_to_js(body, xml_options) {
+    return P.fromCallback(callback => xml2js.parseString(body, xml_options, callback));
```

Comment:
apparently there's promise support already - 
https://github.com/Leonidas-from-XIV/node-xml2js#promise-usage

```suggestion
    return xml2js.parseStringPromise(body, xml_options);
```

Reply:
Done, but a little bit differently. 
Just pulled the actual `parseStringPromise` into a const named `parse_xml_to_js`.